src/jit/morph.cpp

```diff
@@ -13665,7 +13665,7 @@ GenTree* Compiler::fgMorphSmpOpOptional(GenTreeOp* tree)
     GenTree*   op2  = tree->gtOp2;
     var_types  typ  = tree->TypeGet();
 
-    if (GenTree::OperIsCommutative(oper))
+    if (!optValnumCSE_phase && GenTree::OperIsCommutative(oper))
```

Comment:
This fix looks like it would suffer the same issue that #11207 addresses in some other code - - `optValnumCSE_phase` becomes false before we're done relying on correct value numbers; we use the value numbers in assertion prop and range check opt, which run after CSE. You could (maybe?)  craft a test case where the thing with the bad value number is used to index into an array, and get us to incorrectly remove the bonds check, even with this fix (if I'm following correctly). Not sure what would be the best adjustment, seems like you could either have the transform erase the value number or add a flag (`optValueNumbersValid`?) that's like `OptValnumCSE_phase` but stays true until we're done with value numbers...